purchases/src/main/kotlin/com/revenuecat/purchases/storage/FileRepository.kt

```diff
@@ -0,0 +1,197 @@
+package com.revenuecat.purchases.storage
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import com.revenuecat.purchases.LogHandler
+import com.revenuecat.purchases.common.currentLogHandler
+import com.revenuecat.purchases.common.verboseLog
+import com.revenuecat.purchases.utils.DefaultUrlConnectionFactory
+import com.revenuecat.purchases.utils.UrlConnectionFactory
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.NonCancellable
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.net.URI
+import java.net.URL
+import java.security.MessageDigest
+
+/**
+ * A file cache.
+ */
+internal interface FileRepositoryType {
+    /**
+     * Prefetch files at the given urls.
+     * @param urls An array of URL to fetch data from.
+     * @return The Job that is executing the prefetch
+     */
+    fun prefetch(urls: List<URL>)
+
+    /**
+     * Create and/or get the cached file url.
+     * @param url The url for the remote data to cache into a file.
+     * @return The local file's URI where the data can be found after caching is complete.
+     */
+    suspend fun generateOrGetCachedFileURL(url: URL): URI
+}
+
+/**
+ * The file manager is a service capable of storing data and returning the URL where that stored data exists.
+ */
+internal interface LargeItemCacheType {
+    fun generateLocalFilesystemURI(remoteURL: URL): URI?
+    fun cachedContentExists(uri: URI): Boolean
+    fun saveData(data: InputStream, uri: URI)
+}
+
+internal class FileRepository(
+    @get:VisibleForTesting
```

Comment:
Nit: We can add this
```suggestion
    @get:VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
```

Reply:
Isn't that the default?

Reply:
Yeah it’s the default… I guess we did it to make it more explicit and now most of our usages include it, so I would add it for consistency